storage/src/backend/registry.rs

```diff
@@ -519,7 +602,26 @@ impl BlobReader for RegistryReader {
     }
 
     fn try_read(&self, buf: &mut [u8], offset: u64) -> BackendResult<usize> {
+        let option_index = self.mirror_state.index.load();
         self._try_read(buf, offset, true)
+            .map_err(|e| {
+                if option_index.is_some() {
+                    let index = option_index.unwrap();
+                    let state = &self.mirror_state.mirrors[index].state;
+                    let config = &self.mirror_state.mirrors[index].config;
+                    error!(
+                        "read filed, current state: {:?}, mirror config: {:?},  error: {:?}",
+                        state, config, e
+                    );
+                    self.set_mirror_status(index, false);
```

Comment:
If this mirror request failed, how can we retry again later?

Reply:
It will retry `state.retry_limit` times in read() method:
```rust
fn read(&self, buf: &mut [u8], offset: u64) -> BackendResult<usize> {
    let mut retry_count = self.retry_limit();
    let begin_time = self.metrics().begin();

    loop {
        match self.try_read(buf, offset) {
            Ok(size) => {
                self.metrics().end(&begin_time, buf.len(), false);
                return Ok(size);
            }
            Err(err) => {
                if retry_count > 0 {
                    warn!(
                        "Read from backend failed: {:?}, retry count {}",
                        err, retry_count
                    );
                    retry_count -= 1;
                } else {
                    self.metrics().end(&begin_time, buf.len(), true);
                    ERROR_HOLDER
                        .lock()
                        .unwrap()
                        .push(&format!("{:?}", err))
                        .unwrap_or_else(|_| error!("Failed when try to hold error"));
                    return Err(err);
                }
            }
        }
    }
}
```